### ISSUE TITLE
[BUG] ERG not dynamically changing resistance in DKN am3i

### ISSUE THREAD

**Issue by redsted** [cagnulein/qdomyos-zwift#2262]
**Describe the bug**
DKN Am3i in Zwift ERG mode does not change resistance to achieve wattage.

**To Reproduce**
1. Connect qz to DKN bike
2. Open zwift and connect to game (comes up as Wahoo Kickr)
3. Change qz mode to ERG in Bike settings
4. Start workout in Zwift and activate ERG
5. Ride

**Expected behavior**
Resistance of bike should adjust to maintain Wattage. 

**Screenshots**

**Desktop (please complete the following information):**
 - OS: macOS
 - Version 12.7.4

**Smartphone (please complete the following information):**
 - Device: Galaxy s21
 - Android 14

**Append a debug log**

 Follow this guide https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F
[inbound1668133825227143321.txt](https://github.com/cagnulein/qdomyos-zwift/files/14810890/inbound1668133825227143321.txt)
[inbound7443895172970167260.txt](https://github.com/cagnulein/qdomyos-zwift/files/14810891/inbound7443895172970167260.txt)

  
  
**Additional context**

**Comment by cagnulein** [comment]
Thanks I will do after the Easter days!
Happy Easter!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 30 mar 2024 alle 07:41 Adam ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
good news! I already did this! Join the beta https://play.google.com/apps/testing/org.cagnulen.qdomyoszwift and update the app from the store.
Then do some spin into all the resistance levels to calibrate qz and then qz will do the magic! Let me know!

**Comment by redsted** [comment]
Thanks! How do I spin in to all resistance levels? Do I up the resistance
or the gears in the app?

**Comment by cagnulein** [comment]
the same, even manually change the reisstance. qz is collecting pairs of cadence + resistance to store the returning wattage. so more data you fill in qz spinning, more precision you will have from qz! let me know!

**Comment by redsted** [comment]
I originally wasn't able to get the DKN Am3i to connect to the app
correctly. I saw an old ticket that suggested turning on Toorx 3.0
compatibility and Toorx/iConsole bike. This results in no display on the
DKN bike and no manual control. Is this correct?

**Comment by cagnulein** [comment]
yes this is correct. ok so change the resistance from the app directly.
anyway qz will can also learn the erg table while you are actually doing an
erg workout, this means that qz will increase the precision while you are
doing it

let me know!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 31 mar 2024 alle 06:05 Adam ***@***.***> ha
scritto:

**Comment by redsted** [comment]
It seems to be working, thank you!

Just to let you know, I tried the beta version of the app yesterday and
this morning it resulted in some weird behaviour. It was almost as though
the qz app or Bluetooth started taking over my phone. Buttons randomly
being pressed then search bar opening and jargon bring entered in without
me doing anything. I reverted back to stable build and it appears to be
working fine now. Has anyone else on beta experienced same issue?
